# gravity? idk wtf is this honestly
--- original/engine/ragdoll.h	2020-11-30 03:47:28.000000000 +0200
+++ mod/engine/ragdoll.h	2023-02-27 21:29:45.845217384 +0200
@@ -445,7 +445,7 @@
 
 void ragdolldata::move(dynent *pl, float ts)
 {
-    extern const float GRAVITY;
+    extern const float gravity;
     if(collidemillis && lastmillis > collidemillis) return;
 
     int material = lookupmaterial(vec(center.x, center.y, center.z + radius/2));
@@ -467,7 +467,7 @@
     {
         vert &v = verts[i];
         vec dpos = vec(v.pos).sub(v.oldpos);
-        dpos.z -= GRAVITY*ts*ts;
+        dpos.z -= gravity*ts*ts;
         if(water) dpos.z += 0.25f*sinf(detrnd(size_t(this)+i, 360)*RAD + lastmillis/10000.0f*M_PI)*ts;
         dpos.mul(pow((water ? ragdollwaterfric : 1.0f) * (v.collided ? ragdollgroundfric : airfric), ts*1000.0f/ragdolltimestepmin)*tsfric);
         v.oldpos = v.pos;
