# pvpf command makes other players transparent
--- original/engine/rendermodel.cpp	2020-11-30 03:47:28.000000000 +0200
+++ mod/engine/rendermodel.cpp	2021-03-18 20:55:35.564263381 +0200
@@ -39,6 +39,9 @@
 MODELTYPE(MDL_SMD, smd);
 MODELTYPE(MDL_IQM, iqm);
 
+bool pvpcollissions = true;
+ICOMMAND(pvpf, "", (), { pvpcollissions = !pvpcollissions; });
+
 #define checkmdl if(!loadingmodel) { conoutf(CON_ERROR, "not loading a model"); return; }
 
 void mdlcullface(int *cullface)
@@ -1113,7 +1116,7 @@
     if(d!=player && !(anim&ANIM_RAGDOLL)) flags |= MDL_CULL_VFC | MDL_CULL_OCCLUDED | MDL_CULL_QUERY;
     if(d->type==ENT_PLAYER) flags |= MDL_FULLBRIGHT;
     else flags |= MDL_CULL_DIST;
-    if(d->state==CS_LAGGED) fade = min(fade, 0.3f);
+    if(d->state==CS_LAGGED || (d->state==CS_ALIVE && !pvpcollissions)) fade = min(fade, 0.3f);
     else flags |= MDL_DYNSHADOW;
     if(drawtex == DRAWTEX_MODELPREVIEW) flags &= ~(MDL_LIGHT | MDL_FULLBRIGHT | MDL_CULL_VFC | MDL_CULL_OCCLUDED | MDL_CULL_QUERY | MDL_CULL_DIST | MDL_DYNSHADOW);
     rendermodel(NULL, mdlname, anim, o, yaw, pitch, flags, d, attachments, basetime, 0, fade);
